packages/babel-core/src/transformation/block-hoist-plugin.js

```diff
@@ -41,24 +39,14 @@ const blockHoistPlugin = {
   visitor: {
     Block: {
       exit({ node }) {
-        let hasChange = false;
-        for (let i = 0; i < node.body.length; i++) {
-          const bodyNode = node.body[i];
-          if (bodyNode?._blockHoist != null) {
-            hasChange = true;
-            break;
-          }
-        }
-        if (!hasChange) return;
-
-        node.body = sortBy(node.body, function (bodyNode) {
-          let priority = bodyNode?._blockHoist;
-          if (priority == null) priority = 1;
-          if (priority === true) priority = 2;
-
-          // Higher priorities should move toward the top.
-          return -1 * priority;
-        });
+        if (node.body.every(node => node?._blockHoist == null)) return;
+        node.body
+          .filter(node => node?._blockHoist == null)
+          .forEach(node => (node.priority = 1));
```

Comment:
Nope, this isn't a great approach; on a fresh readthrough, `priority` isn't a property that previously existed on these nodes (`priority` in the existing implementation is a purely local variable), so it's not wise to *set* it as a property on them either.